lib/graphql/schema/scrubber.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+require "set"
+
+module GraphQL
+  class Schema
+    # Pass options to `use(SomeTracing, scrub_variables: { ... })`
+    #
+    # @api private
+    class Scrubber
+      # Create a scrubber which filters query variables.
+      #
+      # It only filters variables, not literal values in the query string.
+      #
+      # @param whitelist [Array<String, Symbol>] Argument names to _allow_
+      # @param blacklist [Array<String, Symbol>] Argument names to _hide_
+      # @param mutations [Boolean] If false, _no_ variables to mutations will be shown
+      def initialize(whitelist: nil, blacklist: nil, mutations: nil)
+        @whitelist = whitelist ? whitelist.map(&:to_s).to_set  : nil
+        @blacklist = blacklist ? blacklist.map(&:to_s).to_set : nil
+        if @whitelist && @blacklist
+          raise ArgumentError, "#{self.class.name} supports whitelist: _or_ blacklist:, but not both."
```

Comment:
like we talked earlier should we open this up to:

```
Scrubber.new(->(variable, query, context) { query.mutation? })
```

or something like `ScrubVariables.new(scrubber).scrub`?

Maybe this can stay private for now, and I can refactor this into a public API?

Reply:
🤔  I don't quite understand, what's the difference here? (Different outputs, different use cases?)

Reply:
I'm thinking of ways to allow users to scrub according to their own rules, like`context[:is_secret_query]` :D

Maybe it's not needed though 🤷‍♂️